content/en/docs/ops/configuration/telemetry/envoy-stats/index.md

```diff
@@ -21,30 +21,57 @@ To see the statistics for a pod:
 $ kubectl exec $POD -c istio-proxy -- pilot-agent request GET stats
 {{< /text >}}
 
-See [the Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats)
-for an explanation of the data recorded.
+Envoy collects various kinds of statistics. The following links are Envoy documents about some stats that it collects:
 
-By default, Istio configures Envoy to record minimal statistics.  The default collection
+- [Upstream connection](https://www.envoyproxy.io/docs/envoy/latest/configuration/upstream/cluster_manager/cluster_stats)
+- [Listener](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/stats)
+- [HTTP Connection Manager](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/stats)
+- [TCP proxy](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/network_filters/tcp_proxy_filter#statistics)
+- [Router](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/router_filter.html?highlight=vhost#statistics)
+
+By default, Istio configures Envoy to record minimal statistics. The default collection
 keys are:
 
 - `cluster_manager`
 - `listener_manager`
 - `server`
 - `cluster.xds-grpc`
+- `wasm`
 
 To see the Envoy settings for statistics data collection use
 [`istioctl proxy-config bootstrap`](/docs/reference/commands/istioctl/#istioctl-proxy-config-bootstrap) and follow the
 [deep dive into Envoy configuration](/docs/ops/diagnostic-tools/proxy-cmd/#deep-dive-into-envoy-configuration).
 Envoy only collects statistical data on items matching the `inclusion_list` within
 the `stats_matcher` JSON element.
 
-To Configure Envoy to record statistics for inbound or outbound traffic, add the
-`sidecar.istio.io/statsInclusionPrefixes` annotation to the pod template in the Kubernetes `Deployment`.
-Add the `cluster.outbound` prefix to gather data about outbound traffic activity and circuit breaking.
-To gather data on inbound traffic, add the `listener` prefix.  The sample
-[fortio-deploy.yaml]({{< github_file>}}/samples/httpbin/sample-client/fortio-deploy.yaml)
-shows use of `sidecar.istio.io/statsInclusionPrefixes` with the `cluster.outbound` prefix.
+{{< tip >}}
+Note Envoy stats name highly depends on how Envoy configuration is composed, and thus could tie to Istio control plane implementation detail.
+If you are building dashboard or alert based on Envoy stats, before upgrading your Istio, it is highly recommended to examine the stats in a canary environment.
+{{< /tip >}}
+
+To configure Istio proxy to record additional statistics, you can add [`ProxyConfig.ProxyStatsMatcher`](/docs/reference/config/istio.mesh.v1alpha1/#ProxyStatsMatcher) to your mesh config. For example, to enable stats for circuit breaker, retry, and upstream connections globally, you can specify stats matcher as follow:
```

Comment:
does this require pod restarts, etc?  What can the user expect if they manage to update?

Reply:
Added a note about this. Thanks!